src/interpreter/bytecode-generator.cc

```diff
@@ -1788,9 +1788,20 @@ void BytecodeGenerator::VisitBreakStatement(BreakStatement* stmt) {
   execution_control()->Break(stmt->target());
 }
 
+// [RUN-3317] Shift breakpoint from STMT to STMT->EXPR().
+#define REPLAY_BERAKPOINT_SHIFT(STMT, EXPR) \
+  if ( \
+    recordreplay::IsRecordingOrReplaying("BreakpointShift") && \
+    STMT->EXPR()->position() >= 0 \
+  ) { \
+    builder()->SetExpressionAsStatementPosition(STMT->EXPR()); \
+  } else { \
+    builder()->SetStatementPosition(STMT); \
+  }
```

Comment:
I don't see why this needs to be a macro.  We should avoid macros whenever possible, can you change this to a helper function?